tools/cache_creator/cache_info.h

```diff
@@ -0,0 +1,99 @@
+ *
+ *  Copyright (c) 2021 Google LLC. All Rights Reserved.
+ *
+ *  Permission is hereby granted, free of charge, to any person obtaining a copy
+ *  of this software and associated documentation files (the "Software"), to deal
+ *  in the Software without restriction, including without limitation the rights
+ *  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ *  copies of the Software, and to permit persons to whom the Software is
+ *  furnished to do so, subject to the following conditions:
+ *
+ *  The above copyright notice and this permission notice shall be included in all
+ *  copies or substantial portions of the Software.
+ *
+ *  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ *  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ *  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ *  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ *  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ *  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ *  SOFTWARE.
+ *
+ **********************************************************************************************************************/
+#pragma once
+
+#include "include/binary_cache_serialization.h"
+
+// These Xlib defines conflict with LLVM.
+#undef Bool
+#undef Status
+
+#include "llvm/ADT/ArrayRef.h"
+#include "llvm/ADT/SmallString.h"
+#include "llvm/ADT/SmallVector.h"
+#include "llvm/ADT/StringMap.h"
+#include "llvm/ADT/StringRef.h"
+#include "llvm/Support/Error.h"
+#include "llvm/Support/MemoryBufferRef.h"
+#include "llvm/Support/raw_ostream.h"
+#include <string>
+
+namespace cc {
+
+// Represents printable information about the public Vulkan Pipeline Cache header.
+struct PublicVkHeaderInfo {
+  const vk::PipelineCacheHeaderData *publicHeader;
+  size_t trailingSpaceBeforePrivateBlob;
+};
+
+llvm::raw_ostream &operator<<(llvm::raw_ostream &os, const PublicVkHeaderInfo &info);
+
+// Represents printable information about the private Pipeline Binary Cache header.
+struct BinaryCachePrivateHeaderInfo {
+  const vk::PipelineBinaryCachePrivateHeader *privateHeader;
+  size_t contentBlobSize;
+};
+
+llvm::raw_ostream &operator<<(llvm::raw_ostream &os, const BinaryCachePrivateHeaderInfo &info);
+
+using MD5DigestStr = llvm::SmallString<32>;
+
+// Represents printable information about a Pipeline Binary Cache entry and its location within the cache blob.
+struct BinaryCacheEntryInfo {
+  const vk::BinaryCacheEntry *entryHeader;
+  size_t idx;
+  llvm::ArrayRef<uint8_t> entryBlob;
+  MD5DigestStr entryMD5Sum;
+  llvm::Optional<llvm::StringRef> sourceElfBlobIdentifier;
+};
+
+llvm::raw_ostream &operator<<(llvm::raw_ostream &os, const BinaryCacheEntryInfo &info);
+
+// Given a directory, returns a map from ELF MD5 sums their paths. Non-ELF files are ignored.
```

Comment:
This comment gives the impression that you look at the contents of the file to see if it is an elf file, but you are actaully looking for files with the `.elf` extension regardless of their contents.